From 1fd11d735e596dae4bdaaae4ccac511d04403f22 Mon Sep 17 00:00:00 2001
From: pratik Bhalgat <pratik_bit@codeaurora.org>
Date: Thu, 12 Dec 2013 20:43:59 +0530
Subject: [PATCH] nt32 value is set to uint16 parameter

change the len size to 32bit. and explicit typecast to 16bit.
to avoid static code analysis error.
CRs-fixed: 576040
Change-Id: Iebda14547a0c63ce8b7dd24e1a8f2b06e97103e6
---
 CORE/SME/src/p2p/p2p_Api.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/CORE/SME/src/p2p/p2p_Api.c b/CORE/SME/src/p2p/p2p_Api.c
index 87be42b..2a9ccd3 100644
--- a/CORE/SME/src/p2p/p2p_Api.c
+++ b/CORE/SME/src/p2p/p2p_Api.c
@@ -112,14 +112,21 @@ eHalStatus p2pProcessRemainOnChannelCmd(tpAniSirGlobal pMac, tSmeCmd *p2pRemaino
     len = sizeof(tSirRemainOnChnReq) + pMac->p2pContext.probeRspIeLength;
 #endif
 
-    status = palAllocateMemory(pMac->hHdd, (void**)&pMsg, len );
+    if( len > 0xFFFF )	
+    {	
+       /*In coming len for Msg is more then 16bit value*/	
+       smsLog(pMac, LOGE, FL("  Message length is very large, %d"),	
+            len);	
+       return eHAL_STATUS_FAILURE;	
+    }
+	status = palAllocateMemory(pMac->hHdd, (void**)&pMsg, len );
     if(HAL_STATUS_SUCCESS(status))
     {
         VOS_TRACE(VOS_MODULE_ID_SME, VOS_TRACE_LEVEL_INFO, "%s call", __func__);
         palZeroMemory(pMac->hHdd, pMsg, sizeof(tSirRemainOnChnReq));
         pMsg->messageType = eWNI_SME_REMAIN_ON_CHANNEL_REQ;
-        pMsg->length = len;
-        palCopyMemory( pMac, pMsg->selfMacAddr, pSession->selfMacAddr, sizeof(tSirMacAddr) ); 
+        pMsg->length = (tANI_U16)len;
+	palCopyMemory( pMac, pMsg->selfMacAddr, pSession->selfMacAddr, sizeof(tSirMacAddr) ); 
         pMsg->chnNum = p2pRemainonChn->u.remainChlCmd.chn;
         pMsg->phyMode = p2pRemainonChn->u.remainChlCmd.phyMode;
         pMsg->duration = p2pRemainonChn->u.remainChlCmd.duration;
-- 
1.8.2.1

